### ISSUE TITLE
log output isn't colored when kernel32 AttachConsole is called

### ISSUE THREAD

**Issue by bitbonk** [serilog/serilog-sinks-console#111]
**Does this issue relate to a new *feature* or an existing *bug*?**
Bug

**What version of Serilog Console Sink is affected by this issue?** 
Serilog.Sinks.Console 3.1.1

**What is the target framework and operating system affected by this issue? Please see [target frameworks](https://docs.microsoft.com/en-us/nuget/schema/target-frameworks) & [net standard matrix](https://docs.microsoft.com/en-us/dotnet/standard/net-standard).**
netCore 3.1 on Windows 10

**Please describe the current behaviour you are experiencing?**
We have a .NET Core 3.1 WPF application that uses the default `<OutputType>WinExe</OutputType>` in the csproj because the applicaition should not open a console window when started by default (e.g. from Windows Explorer).
However if the application is started from a Console Window or from an IDE that captures console output (such as JetBrains Rider), the console output of the WPF application should be redirected to that console of the parent process. For that we use [this approach](http://www.csharp411.com/console-output-from-winforms-application/). 
Now this application also uses `Microsoft.Extensions.Hosting` in conjunction with Serilog. 

The problem now is that in the following code the console output is not coloured (you can create a simple WPF app to reproduce it):

```xml
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

    <PropertyGroup>
        <OutputType>WinExe</OutputType>
        <TargetFramework>netcoreapp3.1</TargetFramework>
        <UseWPF>true</UseWPF>
    </PropertyGroup>

    <ItemGroup>
      <PackageReference Include="Serilog.Sinks.Console" Version="3.1.1" />
    </ItemGroup>

</Project>
```

```c#
namespace WpfApp1
{
    using System.Runtime.InteropServices;
    using Serilog;
    using Serilog.Sinks.SystemConsole.Themes;

    public partial class App
    {
        private const int ATTACH_PARENT_PROCESS = -1;

        public App()
        {
            // Create the fallback logger in case anything happens before ConfigureLogging()
            // is called configured by Microsoft.Extensions.Hosting
            // (Remove this code to make it work.)
            Log.Logger = new LoggerConfiguration()
                .Enrich.FromLogContext()
                .WriteTo.Console(theme: AnsiConsoleTheme.Code)
                .CreateLogger();

            ConfigureLogging();

            RunApp();
        }

        private static void RunApp()
        {
            Log.Logger.Error("This is an error that should be displayed red");
            Log.Logger.Information("This is an information that should be displayed bold an black");
        }

        private static void ConfigureLogging()
        {
            // This needs to be called before the first call to Console.WriteLine().
            // We need to call this in ConfigureLogging because in reality it depends on IConfiguration from Microsoft.Extensions
            AttachConsole(ATTACH_PARENT_PROCESS);

            Log.Logger = new LoggerConfiguration()
                .MinimumLevel.Verbose()
                .Enrich.FromLogContext()
                .WriteTo.Console(
                    theme: AnsiConsoleTheme.Code).CreateLogger();
        }
    }
}
```

We have to remove code at the beginning that creates the fallback logger to get the coloured console back. 
Even if we do not log anything at all before we call `AttachConsole` we still need to remove those lines. The fallback logger must not be initialized to a console at all before `AttachConsole` is called.

**Please describe the expected behaviour if the ?**
If we do not log anything before `ConfigureLogging()` is called, the console entries inside `RunApp` should be coloured, also if we have created the fallback logger at the beginning.

**Comment by nblumhardt** [comment]
Does it work for you if you don't apply a theme to the intial (fallback) logger?

Best regards,
Nick

**Comment by bitbonk** [comment]
It surely didn't work if I applied no theme at all (so the theme parameter was null).

My guess is that it won't work with `ConsoleTheme.None` either because when I debugged the decompiled code using Rider and both `Console.IsOutputRedirected` `Console.IsErrorRedirected`  where `true` in the second call of `WriteTo.Console` (after `AttachConsole()`) when the fallback logger was specified. When the fallback logger was _not_ specified, those two properties where `false` in the same call of `WriteTo.Console`.

**Comment by bitbonk** [comment]
I also the that your dev branch has the `applyThemeToRedirectedOutput` parameter which would probably also "solve" the problem when set to `true`.

**Comment by bitbonk** [comment]
Applying `ConsoleTheme.None` for the fallback logger does not solve the problem. The output still is not colored.

```C#
Log.Logger = new LoggerConfiguration()
    .Enrich.FromLogContext()
    .WriteTo.Console(theme: ConsoleTheme.None)
    .CreateLogger();
```

**Comment by nblumhardt** [comment]
Sounds like the `applyThemeToRedirectedOutput` option is the way to go, since it already exists. Have you tried the `-dev` package?

**Comment by nimrod46** [comment]
> Sounds like the `applyThemeToRedirectedOutput` option is the way to go, since it already exists. Have you tried the `-dev` package?

@nblumhardt I can confirm this has fixed the issue, using version `4.0.0-dev-00839` on .Net 5
Well done even the Windows console is colored

**Comment by nblumhardt** [comment]
Thanks @nimrod46 👍
